com.unity.render-pipelines.universal/ShaderLibrary/ShaderVariablesFunctions.hlsl

```diff
@@ -187,13 +201,13 @@ real ComputeFogFactor(float z)
         // -density * z computed at vertex
         return real(unity_FogParams.x * clipZ_01);
     #else
-        return 0.0h;
+        return real(0.0);
     #endif
 }
 
-real ComputeFogIntensity(real fogFactor)
+half ComputeFogIntensity(half fogFactor)
```

Comment:
I particularly don't like to have overloads like this for several functions. Some thoughts:
  - Overloads could be templated similar to: https://github.com/Unity-Technologies/Graphics/blob/master/com.unity.render-pipelines.core/ShaderLibrary/Macros.hlsl#L66-L70
  - Use `real` for the functions? I guess the cast is a problem because we are not using `real` across the board in URP library but we could cast on calling site?